### PR TITLE
Fix #292 -- Change admin widgets' JS media ordering

### LINKED ISSUE #292: i18n is broken due to missing jQuery import in media 🐛
### Bug Description

When using django-select2 in the Django Admin (regardless of the value of settings.LANGUAGE_CODE) the corresponding i18n file (e.g. en.js) throws the following exception in the brower console:

```js
TypeError: Cannot read properties of undefined (reading 'define')
    at en.js:3:106
    at en.js:3:755
```

The problem is that jquery.init.js is loaded before this file. In order to load it afterwards [this line](https://github.com/codingjoe/django-select2/blob/07ea4b8cf996eb587fd3dacf4361bd579d83e8e2/django_select2/forms.py#L158) has to be replaced with:

```python
            js=select2_js + i18n_file + ["admin/js/jquery.init.js"] + ["django_select2/django_select2.js"],
```

### Steps to Reproduce

1. Create the following model:

```python
from django.db import models

class Person(models.Model):
    title = models.CharField(max_length=10, choices=[('Dr.', 'Dr.'), ('Rep.', 'Rep.')])
```

2. Create the following admin for the model:

```python
from django.contrib import admin
from django import forms

from django_select2.forms import Select2Widget

from .models import Person


class PersonForm(forms.ModelForm):
    class Meta:
        widgets = {
            'title': Select2Widget
        }


@admin.register(Person)
class PersonAdmin(admin.ModelAdmin):
    form = PersonForm
```

3. Visit the admin page for the model (admin/app/person/add/)

4. Open the browser console and find the exception thrown in en.js

### Expected Behavior

No exception is thrown and the i18n works as expected

### PR DESCRIPTION
**Pull request by mardukbp**
Select2Mixin now loads admin/js/jquery.init.js, just like the AutocompleteMixin of Django Admin. Fixes #292.

### CODE REVIEW

**Review by codingjoe — APPROVED** [review]

File: django_select2/forms.py [hunk 1]
```diff
@@ -155,7 +155,7 @@ def media(self):
             i18n_file = [f"{settings.SELECT2_I18N_PATH}/{self.i18n_name}.js"]
 
         return forms.Media(
-            js=select2_js + i18n_file + ["django_select2/django_select2.js"],
+            js=select2_js + i18n_file + ["admin/js/jquery.init.js"] + ["django_select2/django_select2.js"],
```
Comment (codingjoe) [inline django_select2/forms.py:158]:
```suggestion
            js=select2_js + i18n_file + ["admin/js/jquery.init.js", "django_select2/django_select2.js"],
```

File: django_select2/forms.py [hunk 2]
```diff
@@ -155,7 +155,7 @@ def media(self):
             i18n_file = [f"{settings.SELECT2_I18N_PATH}/{self.i18n_name}.js"]
 
         return forms.Media(
-            js=select2_js + i18n_file + ["django_select2/django_select2.js"],
+            js=select2_js + i18n_file + ["admin/js/jquery.init.js", "django_select2/django_select2.js"],
```
Comment (codingjoe) [inline django_select2/forms.py:158]:
```suggestion
            js=select2_js
            + i18n_file
            + ["admin/js/jquery.init.js", "django_select2/django_select2.js"],
```

**Comment by codingjoe** [comment]
@mardukbp I made an amendment to your patch and moved the behavior to the admin mixin only. Can you double-check if that works for you?

**Comment by mardukbp** [comment]
@codingjoe It does not work. Why did not you wait for my reply? Why did you modify my PR, when I had already tested that it solves the problem? Please use my original code.

**Comment by codingjoe** [comment]
Hi there,

Don't take this wrong, but I maintain dozens of packages. I don't remember why, I did something 3 weeks ago. Upon reviewing your code again, it wasn't something I would have merged. You modified the base widget that shouldn't implement anything specific to Django's admin. Which you may have noticed because all tests related to that were failing. Thus, your code 

Please check if you are using the admin mixin, that is required to make select2 work in Django's admin. If the error persists, please open a new ticket.

Kindly,
Joe

**Comment by mardukbp** [comment]
Thank you for your prompt and careful reply.  I appreciate that you take the time to respond quickly. I agree that my modification was not done at the right place and the right place is the AdminMixin.  

I think that as a maintainer you should merge PRs only when there's convincing evidence that they solve a problem and not rush to get rid of open PRs. Now we have to start all over again with another issue and another PR, which costs both of us time. In other words, I appreciate your dilligence, but don't be too efficient :)

**Pull request merged**